core/src/main/java/it/pagopa/selfcare/mscore/core/ContractService.java

```diff
@@ -266,8 +266,7 @@ private NotificationToSend toNotificationToSend(Institution institution, Token t
         }
         notification.setNotificationType(queueEvent);
         notification.setFileName(retrieveFileName(token.getContractSigned(), token.getId()));
-        // TODO: persist the proper contentType (related to the contract) in the database, it should be application/json for onboarding made through the autocomplete api and it should be application/octet-stream or the proper type for onboardings made through the frontend
-        notification.setContentType("application/octet-stream");
+        notification.setContentType(token.getContentType() == null ? "application/octet-stream" : token.getContentType());
```

Comment:
should replace "application/octet-stream" with object **MediaType.APPLICATION_OCTET_STREAM_VALUE**, it is better avoid string hardcoded